packages/esm-appointments-app/src/config-schema.ts

```diff
@@ -69,6 +82,63 @@ export const configSchema = {
       'Whether to show the Unscheduled Appointments tab. Note that configuring this to true requires a custom unscheduledAppointment endpoint not currently available',
     _default: false,
   },
+
+  appointmentTables: {
+    _description: 'Configuration for appointment tables',
+    columnDefinitions: {
+      _type: Type.Array,
+      _default: [],
+      _description: 'Custom columns for appointment tables',
+      _elements: {
+        _validators: [
+          validator(
+            (columnDef: AppointmentColumnDefinition) =>
+              Boolean(columnDef.columnType || appointmentColumnTypes.some((c) => c == columnDef.id)),
+            (columnDef) =>
+              `No columnType provided for column with ID '${
+                columnDef.id
+              }', and the ID is not a valid columnType. Valid column types are: ${appointmentColumnTypes.join(', ')}.`,
+          ),
+        ],
+        id: {
+          _type: Type.String,
+          _description: 'The unique identifier for the column',
+        },
+        columnType: {
+          _type: Type.String,
+          _description: 'The type of column, if different from the ID',
+          _validators: [validators.oneOf(appointmentColumnTypes)],
+          _default: null,
+        },
+        header: {
+          _type: Type.String,
+          _description: 'The header text for the column',
+          _default: null,
+        },
+        config: {
+          _type: Type.Object,
+          _description: 'Column-specific configuration',
+          _default: {},
+        },
+      },
+    },
+    tableDefinitions: {
```

Comment:
Oh wait, I misread this. This is an array of tables. Why? And then the hook just combines all the columns? This PR has so much unnecessary complexity. Please do more careful work when using AI tools; it is very easy for them to generate lots of useless crap.